### ISSUE TITLE
Support '/' in branch names

### ISSUE THREAD

**Issue by nielstenboom** [metcalfc/changelog-generator#179]
Hi there!

Wanted to try out your action. I set the head-ref to the current branch which has a `/` in it and got the following error:

```
Error: Branch names must contain only numbers, strings, underscores, periods, and dashes.
```

Would it be possible to support branch names with slashes? Thanks!

**Comment by metcalfc** [comment]
Hey,

I dug into this. There might be a way but I don't have the git internals knowledge to get it to work. So here is what is happening:

* GitHub Actions are shallow checkouts of just the branch in question
* The action "unshallows" the checkout so we've got access to all the tags and branch names.
https://github.com/metcalfc/changelog-generator/blob/baff252eae625e71a89b7c54ad68f51420a06935/changelog.sh#L16-L18
* Branches with slashes are never realized in that. I'm not sure why. 

If you have the magic incantation needed to unshallow a checkout and pull in branches with slashes happy to fix it.

To test this I set a branch `test/branch` to v0.0.2 and ran the action. This is the error: 

```
fatal: ambiguous argument ''v0.0.1...test/branch'': unknown revision or path not in the working tree.
```

So one would probably have to check if the refs have slashes and then pull all the refs in that directory but this is beyond my git foo.

**Comment by metcalfc** [comment]
Just in case someone wants to pick up where I left off https://github.com/metcalfc/changelog-generator/tree/slash_branch has the last patch I tried.

**Comment by nielstenboom** [comment]
Thanks for having a look! 

I took a stab at getting it to work as well and got it to work (sorta.. 😅 )

It can work with 2 gotchas:
1. Have the checkout action do all the fetching lifting, so add `fetch-depth: 0` and don't have this action do any fetching (see [here](https://github.com/nielstenboom/changelog-generator/blob/main/changelog.sh) for the bash file)
2. Add `origin/` in front of the ref that has a `/` in it, with that the `git log ... ` works (no clue about the git foo underneath)

See [here](https://github.com/nielstenboom/recurring-content-detector/actions/runs/4096178019/jobs/7063630642) for a succesful run.

So to fix this we could maybe add a parameter to this action to enable fetching or not, sth like `fetch: false` and then in the readme add that you should put origin in front of branch name refs with forward slashes in them?

**Comment by metcalfc** [comment]
I'm good with the parameter to fetch or not. We can default fetch to true so it doesn't break existing users. You and others that want to use slash or other things can set fetch to false and everyone is off to the races.

**Comment by metcalfc** [comment]
Merged in #188
